share/spice/xkcd/display/xkcd_display.js

```diff
@@ -1,5 +1,39 @@
 (function(env){
     "use strict";
+    
+    function build_comic_number(number) {
+        var comic_number_string = "";
+        if (number) {
+            comic_number_string = "#" + number + " created on "
+        }
+        
+        return comic_number_string;
+    }
+    
+    function build_created_date(month, day, year) {
+        var created_date_string = "";
+        if (month) {
+            created_date_string += month + " ";
+        }
+        if (day) {
+            created_date_string += day + ", ";
+        }
+        if (year) {
+            created_date_string += year;
+        }
+        
+        return created_date_string
+    }
+    
+    function build_comic_number_and_created_date_string(number, month, day, year) {
```

Comment:
The only thing is that this code feels pretty verbose. I don't think we need to check if each parameter is defined and everything. It's no problem, though, at least it's clear what's happening :smiley: